### DISCUSSION TITLE
Warnings when `targets::tar_load()` but I cannot make a reprex

### DISCUSSION THREAD

**Discussion by psychelzh** [ropensci/targets#433] [Help] [answered]
When I call `targets::tar_load()`, the object is loaded correctly, but there is always a warning (following {reprex} is from my working directory):

*Local `.Rprofile` detected at `C:\Users\liang\OneDrive\Documents\Research\cogstruct\.Rprofile`*

``` r
targets::tar_load(config_game)
#> Warning in file_diff_dbl(chr): NAs introduced by coercion
```

<sup>Created on 2021-04-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessionInfo()
#> R version 4.0.5 (2021-03-31)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 19042)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=Chinese (Simplified)_China.936 
#> [2] LC_CTYPE=Chinese (Simplified)_China.936   
#> [3] LC_MONETARY=Chinese (Simplified)_China.936
#> [4] LC_NUMERIC=C                              
#> [5] LC_TIME=Chinese (Simplified)_China.936    
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices datasets  utils     methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] igraph_1.2.6      knitr_1.32        magrittr_2.0.1    tidyselect_1.1.0 
#>  [5] R6_2.5.0          rlang_0.4.10      fansi_0.4.2       stringr_1.4.0    
#>  [9] styler_1.4.1      highr_0.9         targets_0.4.1     tools_4.0.5      
#> [13] data.table_1.14.0 xfun_0.22         utf8_1.2.1        cli_2.4.0        
#> [17] withr_2.4.2       htmltools_0.5.1.1 ellipsis_0.3.1    yaml_2.2.1       
#> [21] digest_0.6.27     tibble_3.1.1      lifecycle_1.0.0   crayon_1.4.1     
#> [25] processx_3.5.1    callr_3.7.0       purrr_0.3.4       codetools_0.2-18 
#> [29] ps_1.6.0          vctrs_0.3.7       fs_1.5.0          glue_1.4.2       
#> [33] evaluate_0.14     rmarkdown_2.7     reprex_2.0.0      stringi_1.5.3    
#> [37] compiler_4.0.5    pillar_1.6.0      backports_1.2.1   renv_0.13.2      
#> [41] pkgconfig_2.0.3
```

</details>

I have no idea how to reproduce it in a new project, but I am sure it is there after upgrading {targets} to the newer version of `0.4.1`.

**Comment by wlandau** [comment, marked as answer]
It's probably the upgrade. It should go away for targets your run with 0.4.1. For now, you can ignore those warnings.

**Reply by psychelzh** [reply]
Thanks!

**Reply by petrbouchal** [reply]
@wlandau just to note I am seeing this using current main branch of targets.

**Reply by wlandau** [reply]
@petrbouchal, is your situation similar to @psychelzh, i.e. migrating away from an old version of `targets`?

**Reply by petrbouchal** [reply]
Thanks @wlandau. Yes, upgraded from 0.4.1 to 0.6.0/0.6.0.9000.

**Reply by jarauh** [reply]
I have the same problem with targets-0.10.0, and I did not upgrade recently. Using ` debug(targets:::file_diff_dbl)`, I can see what the problem is, at least for me. The input is of the following form:
```
[61] NA                   "t19074.4822615773s" "t19074.4817933748s" "t19074.48231149s"   "t19074.4827771608s"
[66] "t19074.4828071807s" NA                   "t19074,4829786199s" "t19074,483772159s"  "t19074,4839215354s"
[71] NA 
```
So, it appears that some numbers are printed with commas instead of points. It seems that my meta-information is corrupted in this way. @wlandau Does that information help?

**Reply by wlandau** [reply]
That's strange. Do you have a reprex? Was the project ever transferred between computers?

**Reply by jarauh** [reply]
No, I don't switch computers.

Maybe the following details can help: The last target runs `rmarkdown::render` on a `.Rmd`-file that sets `options(OutDec = ",")`. All times of that last target are corrupted (but apparently none of the other times). With this option, `as.numeric(as.character(1.2))` and the like fail. So my assumption is that target at some place writes the time (which internally is a numeric) as a character taking into account `OutDec`.

I will try to setup a reprex.

**Reply by wlandau** [reply]
Ah, knowing about `OutDec` does help. Thanks! I suspect `targets` will just need to temporarily set it to `"."` in one of the methods.

**Reply by jarauh** [reply]
Ok, I posted the reprex as a new answer. If it helps, I can also post a bug report.

**Comment by jarauh** [comment]
Here comes the reprex. I adjusted the file `_targets` from the minimal example [targets-minimal](https://github.com/wlandau/targets-minimal):
```r
library(targets)
library(tarchetypes)
source("R/functions.R")
options(tidyverse.quiet = TRUE)
tar_option_set(packages = c("dplyr", "ggplot2", "readr", "tidyr"))
list(
  tar_target(
    raw_data_file,
    "data/raw_data.csv",
    format = "file"
  ),
  tar_target(
    raw_data,
    read_csv(raw_data_file, col_types = cols())
  ),
  tar_target(
    data,
    {
      options(OutDec = ",")
      raw_data %>%
        filter(!is.na(Ozone)) %>% head()
    }
  ),
  tar_target(hist, create_plot(data)),
  tar_target(fit, lm(Ozone ~ Wind + Temp, data)),
  tar_render(report, "index.Rmd")
)
```
Actually, I think that `targets` should not change the option, but rather use the corresponding parameter of `format` or so. But I'm not familiar with the internals of `targets`.

**Reply by wlandau** [reply]
Thanks, should be fixed for new pipelines in fb3f9f6cf96d58b38ca22406d270565f3ad04fbc. `withr::local_options()` temporarily modifies `OutDec` so the option is set back to its original state when the calling function exits.

**Reply by jarauh** [reply]
Great, thanks. That was quick!

**Comment by tjmahr** [comment]
Just chiming here because I had this warning and solved it by clearing out a dead target. It seems like one of the hash values had landed in the `time` column for a target:

```
m_raw <- readr::read_delim("_targets/meta/meta") 
m_raw[77, ] |> glimpse()

Rows: 1
Columns: 18
$ name       <chr> "file_rate_means"
$ type       <chr> "stem"
$ data       <chr> "049b9b6f9ff3148c"
$ command    <chr> "443fc55068e50d6b"
$ depend     <chr> "ef46db3751d8e999"
$ seed       <dbl> -1279112270
$ path       <chr> "data/rate-means-for-export.csv"
$ time       <chr> "eb429828efa39012"
$ size       <chr> "f1f5b67ba969b5ce"
$ bytes      <dbl> 410479
$ format     <chr> "file"
$ repository <chr> "local"
$ iteration  <chr> "vector"
$ parent     <lgl> NA
$ children   <lgl> NA
$ seconds    <dbl> 0
$ warnings   <chr> NA
$ error      <lgl> NA
```

Which led to these warnings for me whenever `tar_meta()` or `tar_read()` were called. The target described by the metadata didn't actually exist anymore (I had renamed it in the pipeline), so using `tar_prune()` removed the bad metadata entry and fixed the warning for me.
